### PR TITLE
Update crates & fix crate vulnerability

### PR DESCRIPTION
**Pull request by dfunkt**
- fixes GHSA-q445-7m23-qrmw by updating openssl to version 0.10.66

### CODE REVIEW

**Review by BlackDex — CHANGES_REQUESTED** [review]
Thanks for the PR/
There are a few crates which also could be updated, update those versions and run `crago update` again.

A nice VSCode extension is `Dependi` which is able to show which versions are the latest version of a specific crate. You still need to verify if you can update those of course.

Also the `cargo-outdated` crate is a nice tool to use, you could run it like this `cargo outdated -R` which would show updated versions of crates which might need code adjustments, but that is not always the case for some crates.

File: Cargo.toml [hunk 1]
```diff
@@ -67,14 +67,14 @@ dashmap = "6.0.1"
 
 # Async futures
 futures = "0.3.30"
-tokio = { version = "1.38.0", features = ["rt-multi-thread", "fs", "io-util", "parking_lot", "time", "signal", "net"] }
+tokio = { version = "1.38.1", features = ["rt-multi-thread", "fs", "io-util", "parking_lot", "time", "signal", "net"] }
```
[the remark this answers is not in the document]
Comment (BlackDex) [reply Cargo.toml:70]: They seem to have yanked 1.39.0, so never mind then.

**Comment by dfunkt** [comment]
Updated the other crates as well this time (including Tokio to 1.39.1)
Only issue I encountered was the update to 6.0.0 of the handlebars crate (due to this commit https://github.com/sunng87/handlebars-rust/commit/598dbd1cb97881bca6cd5d193f1873889d743c08):

```
error[E0639]: cannot create non-exhaustive struct using struct expression
    --> src/config.rs:1339:9
     |
1339 | /         DirectorySourceOptions {
1340 | |             tpl_extension: ".hbs".to_owned(),
1341 | |             ..Default::default()
1342 | |         },
     | |_________^
```
Not sure if I fixed it properly/the right way, still pretty new to Rust :)

**Comment by BlackDex** [comment]
Yes, the solution is good. Thanks!

**Review by BlackDex — APPROVED** [review]

**Review by dani-garcia — APPROVED** [review]

**Pull request merged**